src/pe/exception.rs

```diff
@@ -0,0 +1,815 @@
+    ///
+    /// This unwind code always appears in a dummy prolog, which is never actually executed but
+    /// instead appears before the real entry point of an interrupt routine, and exists only to
+    /// provide a place to simulate the push of a machine frame. This operation records that
+    /// simulation, which indicates the machine has conceptually done this:
+    ///
+    ///  1. Pop RIP return address from top of stack into `temp`
+    ///  2. `$ss`, Push old `$rsp`, `$rflags`, `$cs`, `temp`
+    ///  3. If error flag is `true`, push the error code
+    ///
+    /// Without an error code, RSP was incremented by `40` and the following was frame pushed:
+    ///
+    /// Offset   | Value
+    /// ---------|--------
+    /// RSP + 32 | `$ss`
+    /// RSP + 24 | old `$rsp`
+    /// RSP + 16 | `$rflags`
+    /// RSP +  8 | `$cs`
+    /// RSP +  0 | `$rip`
+    ///
+    /// With an error code, RSP was incremented by `48` and the following was frame pushed:
+    ///
+    /// Offset   | Value
+    /// ---------|--------
+    /// RSP + 40 | `$ss`
+    /// RSP + 32 | old `$rsp`
+    /// RSP + 24 | `$rflags`
+    /// RSP + 16 | `$cs`
+    /// RSP +  8 | `$rip`
+    /// RSP +  0 | error code
+    PushMachineFrame(bool),
+
+    /// A reserved operation without effect.
+    Noop,
+}
+
+/// Context used to parse unwind operation.
+#[derive(Clone, Copy, Debug, PartialEq)]
+struct UnwindOpContext {
+    /// Version of the unwind info.
+    version: u8,
+
+    /// The nonvolatile register used as the frame pointer of this function.
+    ///
+    /// If this register is non-zero, all stack frame offsets used in unwind operations are of type
+    /// `StackFrameOffset::FP`. When loading these offsets, they have to be based off the value of
+    /// this frame register instead of the conventional RSP. This allows the RSP to be modified.
+    frame_register: Register,
+}
+
+/// An unwind operation that is executed at a particular place in the function prolog.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct UnwindCode {
+    /// Offset of the corresponding instruction in the function prolog.
+    ///
+    /// To be precise, this is the offset from the beginning of the prolog of the end of the
+    /// instruction that performs this operation, plus 1 (that is, the offset of the start of the
+    /// next instruction).
+    ///
+    /// Unwind codes are ordered by this offset in reverse order, suitable for unwinding.
+    pub code_offset: u8,
+
+    /// The operation that was performed by the code in the prolog.
+    pub operation: UnwindOperation,
+}
+
+impl<'a> TryFromCtx<'a, UnwindOpContext> for UnwindCode {
+    type Error = error::Error;
+    type Size = usize;
+
+    #[inline]
+    fn try_from_ctx(
+        bytes: &'a [u8],
+        ctx: UnwindOpContext,
+    ) -> Result<(Self, Self::Size), Self::Error> {
+        let mut read = 0;
+        let code_offset = bytes.gread_with::<u8>(&mut read, scroll::LE)?;
```

Comment:
Yes, to my knowledge the only platform using big-endian was Xbox 360 and even then PEs were in little-endian (except for the code and data). However, since the rest of the PE code assumes little-endian too, I've opted for streamlining to the rest of the code base.